specs/sharding/beacon-chain.md

```diff
@@ -424,21 +424,17 @@ def get_start_shard(state: BeaconState, slot: Slot) -> Shard:
     Return the start shard at ``slot``.
     """
     current_epoch_start_slot = compute_start_slot_at_epoch(get_current_epoch(state))
-    shard = state.current_epoch_start_shard
-    if slot > current_epoch_start_slot:
+    committee_count = get_committee_count_per_slot(state, compute_epoch_at_slot(slot))
+    active_shard_count = get_active_shard_count(state, compute_epoch_at_slot(slot))
+    if slot >= current_epoch_start_slot:
         # Current epoch or the next epoch lookahead
-        for _slot in range(current_epoch_start_slot, slot):
-            committee_count = get_committee_count_per_slot(state, compute_epoch_at_slot(Slot(_slot)))
-            active_shard_count = get_active_shard_count(state, compute_epoch_at_slot(Slot(_slot)))
-            shard = (shard + committee_count) % active_shard_count
+        slots_diff = slot - current_epoch_start_slot
+        return (state.current_epoch_start_shard + slots_diff) % active_shard_count
     elif slot < current_epoch_start_slot:
         # Previous epoch
-        for _slot in list(range(slot, current_epoch_start_slot))[::-1]:
```

Comment:
This `_slot` is actually different, it's there for a reason. I agree that the previous code is complicated and not pretty though. Maybe we can optimize it in a different way?

Reply:
Yep, totally right! The slot could be further than 1 epoch from the current state epoch. 
Then those changes are incorrect. Leaving as-is